modules/transports/core/nhttp/src/main/java/org/apache/synapse/transport/passthru/AbstractStreamInterceptor.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright (c) 2021, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.synapse.transport.passthru;
+
+import org.apache.axis2.context.MessageContext;
+
+import java.nio.ByteBuffer;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Abstract class implementing the methods of {@link StreamInterceptor}. A stream interceptor can be written by
+ * extending and overding the necessary methods of this class.
+ * For an example see{@link SampleStreamInterceptor}
+ */
+public abstract class AbstractStreamInterceptor implements StreamInterceptor {
+
+    private String name;
+    private Map<String, Object> properties = new HashMap<String, Object>();
```

Comment:
Would java.util.Properties be a better pick? Can concurrency be an issue here?

Reply:
fixed